### PR TITLE
CHI-2119 - 1 - Add conversationMedia to HRM contact and url_provider block to task

### PR DESCRIPTION
**Pull request by robert-bo-davis**
## Description

This adds the conversationMedia attribute for eternally stored recordings to the contact that is sent to HRM.

It also add the url_provider attribute to the conversations.media task attributes for use by the Insights player.

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Related Issues
Fixes CHI-1099, CHI-1346, CHI-1760

### Verification steps

Run tests in #1573

### CODE REVIEW

File: plugin-hrm-form/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "dev:local-form-definition-server": "npx http-server ../hrm-form-definitions/form-definitions -p 8090 --cors -c-1",
     "dev": "cross-env UNBUNDLED_REACT=true NO_MONITORING=true REACT_FORM_DEFINITIONS_BASE_URL=http://localhost:8090 run-p dev:local-form-definition-server start",
     "dev:s3-forms": "cross-env UNBUNDLED_REACT=true NO_MONITORING=true npm start",
+    "dev:local-serverless": "cross-env REACT_SERVERLESS_BASE_URL=http://localhost:3030 npm run dev",
```
Comment (robert-bo-davis) [inline plugin-hrm-form/package.json:17]: Improve naming here

File: plugin-hrm-form/src/hrmConfig.ts [hunk 2]
```diff
@@ -43,6 +44,8 @@ const readConfig = () => {
   const assetsBucketUrl = manager.serviceConfiguration.attributes.assets_bucket_url;
   const getFormDefinitionsBaseUrl = buildFormDefinitionsBaseUrlGetter(new URL(configuredFormDefinitionsBaseUrl));
 
+  const externalRecordingsEnabled = manager.serviceConfiguration.attributes.external_recordings_enabled || false;
```
Comment (robert-bo-davis) [inline plugin-hrm-form/src/hrmConfig.ts:47]: Have large discussion with the team around the distinction between featureFlags and configuration options and how to represent those in the service configuration.

File: plugin-hrm-form/src/services/ContactService.ts [hunk 3]
```diff
@@ -183,7 +205,71 @@ export function transformForm(form: TaskEntry, conversationMedia: ConversationMe
...
+    returnData.conversationMedia.push({
+      store: 'twilio',
+      reservationSid: task.sid,
+    });
+  }
+
+  const externalRecordingInfo = await getExternalRecordingInfo(task);
+  if (isFailureExternalRecordingInfo(externalRecordingInfo)) {
+    throw new Error(`Error getting external recording info: ${externalRecordingInfo.error}`);
+  }
```
Comment (GPaoloni) [inline plugin-hrm-form/src/services/ContactService.ts:247]: Personal preference here: calling the function only if conditions are met might be clearer? Non blocking personal preference, safe to ignore!

**Review by GPaoloni — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**